src/stores/seismic-simulation-store.ts

```diff
@@ -383,6 +390,13 @@ export const SeismicSimulationStore = types
     },
     setPlotOnGraph(){
       const lastGroup = self.deformationHistory[self.deformationHistory.length - 1];
+
+      const firstValueOfLastGroup = lastGroup.values[0];
+
+      if (firstValueOfLastGroup.plotOnGraph === false){
```

Comment:
A small comment here about intention could be helpful for someone reading without much context.